pkg/controller/inference/playground_controller.go

```diff
@@ -85,18 +86,11 @@ func (r *PlaygroundReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	service := &inferenceapi.Service{}
 	if err := r.Get(ctx, types.NamespacedName{Name: playground.Name, Namespace: playground.Namespace}, service); err == nil {
 		if !controllerutil.HasControllerReference(service) {
-			condition := metav1.Condition{
-				Type:    inferenceapi.PlaygroundProgressing,
-				Status:  metav1.ConditionFalse,
-				Reason:  "AbortProcessing",
-				Message: "Playground owns the same name with an existing Service",
+			err = fmt.Errorf("the Playground %v owns the same name with an exist Service", klog.KObj(playground))
+			if changed := handleUnexpectedCondition(playground, true, false); changed {
+				err = r.Client.Status().Update(ctx, playground)
 			}
-			apimeta.SetStatusCondition(&playground.Status.Conditions, condition)
-			if err := r.Client.Status().Update(ctx, playground); err != nil {
-				return ctrl.Result{}, err
-			}
-			// No need to requeue. Once the Service is deleted, the Playground will be reconciled.
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, err
```

Comment:
If the service is not owned by the playground, we don't need to put it into the queue again.